### ISSUE TITLE
[Bug]: Uncaught DOMException: Background Sync is disabled.

### ISSUE THREAD

**Issue by momar1981** [electron/electron#31063]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

13.1.8

### What operating system are you using?

Windows

### Operating System Version

Windows 10

### What arch are you using?

x64

### Expected Behavior

Electron has background sync disabled. Background sync is used to trigger the service worker to sync changes. It helps to sync work which was done by user when he was offline. Would be good to enable it by default. It is enabled by default in chromium and chrome.
On page with service worker registered, this snippet should produce no errors.

```
navigator.serviceWorker.ready.then(function(swRegistration) {
  return swRegistration.sync.register('myFirstSync');
});
```

This issue was raised at [#9802](https://github.com/electron/electron/issues/9802) 
It is unclear to me why [#9802](https://github.com/electron/electron/issues/9802) was merged with [#6697](https://github.com/electron/electron/issues/6697). The other issue is about Web Push, which is distinct from Background Sync.
Also, the [#6697](https://github.com/electron/electron/issues/6697) was closed without a fix to the Background Sync problem [#9802](https://github.com/electron/electron/issues/9802) .

### Actual Behavior

When run with electron, I get

Uncaught (in promise) DOMException: Background Sync is disabled.

### Additional Information

**How to reproduce**
Open youtube.com with electron

OR

git clone https://github.com/hokein/electron-sample-apps
cd electron-sample-apps
npm install -g electron
electron service-worker/mock-response
Open the console and type in the snippet:

navigator.serviceWorker.ready.then(function(swRegistration) {
  return swRegistration.sync.register('myFirstSync');
});
It gives the error in electron. Use the same snippet in youtube on the browser, it runs fine.

**Comment by codebytere** [comment]
Gist Version: https://gist.github.com/6055066667672505f20edfb1d53a52b7

**Comment by momar1981** [comment]
yes .. this issue still affecting my business... could you please reopen it .

**Comment by abdallahemad94** [comment]
the issue still not fixed on version 26.2., any plans for it
